web/client/main.js

```diff
@@ -207,3 +249,37 @@ function log(msg) {
     log.div.appendChild(document.createTextNode(msg));
     log.div.appendChild(document.createElement("br"));
 }
+
+// setup chat
+function setupChat() {
+    channel.onopen = function () {
```

Comment:
Why setup chat and then check if the user wants to use chat? Also, please check for DataChannel support.